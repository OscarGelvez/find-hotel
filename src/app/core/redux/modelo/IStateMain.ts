export interface IStateMain {
  isLoading: boolean;
}
