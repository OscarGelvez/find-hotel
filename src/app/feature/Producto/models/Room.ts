export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
