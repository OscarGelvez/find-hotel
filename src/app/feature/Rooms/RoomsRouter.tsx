import { LazyFallback } from 'app/shared/components/LazyFallback';
import * as React from 'react';
import { Route, Switch } from 'react-router-dom';

const MainPage = React.lazy(() => import('./pages/Main'));
const DetailPage = React.lazy(() => import('./pages/RoomDetail'));

export const RoomsRouter = () => (
  <React.Suspense fallback={<LazyFallback />}>
    {/* Layout compartido entre las rutas va aquÃ­ */}
    <Switch>
      <Route path="/rooms/detail" component={DetailPage}></Route>
      <Route path="/rooms" component={MainPage}></Route>
    </Switch>
  </React.Suspense>
);
