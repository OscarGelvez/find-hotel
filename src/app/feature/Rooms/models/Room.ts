export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
