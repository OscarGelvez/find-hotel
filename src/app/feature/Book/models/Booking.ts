import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
