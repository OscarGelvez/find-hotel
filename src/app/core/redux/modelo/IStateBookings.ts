import { IBooking } from '../../../feature/Book/models/Booking';

export interface IStateBookings {
  listBookings: IBooking[];
  deleteBookingId: number;
}
