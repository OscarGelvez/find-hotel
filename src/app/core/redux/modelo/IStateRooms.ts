import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IStateRooms {
  listRooms: IRoom[];
  selectedRoomId: number;
}
