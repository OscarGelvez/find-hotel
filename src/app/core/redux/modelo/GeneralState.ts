import { IStateBookings } from './IStateBookings';
import { IStateRooms } from './IStateRooms';
import { IStateMain } from './IStateMain';

export interface Istate {
  rooms: IStateRooms;
  bookings: IStateBookings;
  main: IStateMain;
}
