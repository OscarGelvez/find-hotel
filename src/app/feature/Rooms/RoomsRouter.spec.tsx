import { IStateRooms } from '../../core/redux/modelo/IStateRooms';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';

import { RoomsRouter } from './RoomsRouter';
import { IRoom } from './models/Room';
const mockStore = configureStore([]);
import { IStateMain } from '../../core/redux/modelo/IStateMain';

import { roomInfo } from '../../shared/utils/data';
import renderer from 'react-test-renderer';

const newRoom: IRoom = roomInfo;
const listRooms: IRoom[] = [newRoom];

const rooms: IStateRooms = {
  listRooms,
  selectedRoomId: -1,
};

const main: IStateMain = {
  isLoading: false,
  errorMessage: { message: '', type: '' },
};
let store;
const history = createMemoryHistory();

test('Prueba a componente Router', () => {
  store = mockStore({
    rooms,
    main,
  });

  const element = renderer
    .create(
      <Provider store={store}>
        <Router history={history}>
          <RoomsRouter />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(element).toMatchSnapshot();
});
