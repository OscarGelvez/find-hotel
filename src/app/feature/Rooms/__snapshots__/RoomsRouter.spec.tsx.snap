// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a componente Router 1`] = `null`;
